api/controllers/location.js

```diff
@@ -0,0 +1,141 @@
+'use strict'
+
+const mongoose = require('mongoose')
+require('../models/Sample') // Register model
+const Sample = mongoose.model('Sample')
+
+const DEFAULT_RSSI = 0
+const K1 = 10
+const K2 = 10
+
+module.exports = {
+  getLocation: async (req, res) => {
+    const buildingId = req.body._id
+    const query = Sample.find({ buildingId: buildingId })
+    query.lean()
+
+    try {
+      const samples = await query.exec()
+
+      if (samples === null) {
+        return res.status(404).json({})
+      }
+
+      // Do algorithm
+      const mainFingerprintSortedSSIDs = sortSSIDs(req.body.fingerprint)
+      samples.forEach((sample) => {
+        sample.sortedIds = sortSSIDs(sample.fingerprint)
+      })
+
+      // Calculate the floor:
+
+      // Strep 1: We don't need it now because we are now receiving the building in the params. But // TODO change this to calculate building
+      // Build R’, a subset of R, with all the samples where the building is b (the building estimated in the previous step)
+
+      // Strep 2:
+      // Build R’’, a subset of R’, with all the samples where the strongest AP is equal to AP0, AP1 or AP2
+      const strongestSSIDs = mainFingerprintSortedSSIDs.slice(0, 3)
+      var R2 = []
+      samples.forEach((sample) => {
+        if (strongestSSIDs.includes(sample.sortedIds[0])) R2.push(sample)
+      })
+
+      // Step 3: TODO we are not doing this so we take it as if #(R'') is always big enough
+      // If #(R’’) < n, then R’’ = R’, where #(.) denotes the cardinality of a set, and n is a parameter.
+
+      // Step 4:
+      // Compute the similarity, S(), between the fingerprint given and all the fingerprints in R’’.
+      // The similarity function S() is the Manhattan distance
+
+      R2.forEach((sample) => {
```

Comment:
```suggestion
      R2.forEach(sample => {
```

Cuando el lambda recibe un solo parámetro no hacen falta los paréntesis (no te lo corrijo en los demás lugares)
Dato de color, cuando no toma parámetros hay que escribir `() =>` con los paréntesis igual

Reply:
(+1)